Improving lua script with args and values

Mzack9999 opened the pull request:
## Proposed changes
Closes #4790 


## Checklist
- [ ] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Extended Redis Lua script execution to accept optional keys and arguments inputs while remaining compatible with prior call patterns.
* **Bug Fixes**
  * Improved Redis connectivity behavior by using the provided execution context for ping.
* **Tests**
  * Added a new integration test covering Redis Lua script execution.
  * Reduced flakiness in headless UI “wait visible” checks by adjusting test timing and timeouts.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Mzack9999 committed:
improving lua script with args and values

Mzack9999 committed:
Merge branch 'dev' into feat-4790-lua

Mzack9999 committed:
fixing comments

Mzack9999 committed:
keys to string

dwisiswant0 commented on pkg/js/libs/redis/redis.go:
```diff
@@ -199,8 +202,45 @@ func RunLuaScript(ctx context.Context, host string, port int, password string, s
+				}
+			}
+		}
+	}
```
I think it's worth to address.

Ref: https://redis.io/docs/latest/develop/programmability/eval-intro/#interacting-with-redis-from-a-script

Mzack9999 committed:
fixing context

Mzack9999 committed:
Merge branch 'dev' into feat-4790-lua

Mzack9999 committed:
fix(test): drop log() calls inflating redis-lua-script result count

The two log() calls each emitted a [JS] stdout line that the integration
harness counted as a result, yielding 3 results vs the expected 1. The
matcher reads `response` from the final expression value (not stdout), so
returning the get RunLuaScript call directly keeps the assertion intact
with a single result.

dwisiswant0 commented:
Test failed,

Mzack9999 committed:
fix flaky

Mzack9999 committed:
Merge branch 'dev' into feat-4790-lua

Mzack9999 committed:
harden sdk tests

Avoid shared templates-dir update races across parallel go test packages and ignore known leveldb/ratelimit/memguardian goroutines in goleak.

Mzack9999 committed:
fix sdk templates

Keep goleak ignores, drop DisableUpdateCheck so CI can install templates, and serialize UpdateIfOutdated across processes.

Mzack9999 committed:
Merge branch 'dev' into feat-4790-lua
